File: board-back/src/main/java/com/example/boardback/repository/BoardRepository.java
package com.example.boardback.repository;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.stereotype.Repository;

import com.example.boardback.entity.BoardEntity;
import com.example.boardback.repository.resultSet.GetBoardResultSet;

@Repository
public interface BoardRepository extends JpaRepository<BoardEntity, Integer>{

  boolean existsByBoardNumber(Integer boardNumber);

  BoardEntity findByBoardNumber(Integer boardNumber);
 
  // description : ?1 => 는 매개변수 boardNumber의 첫번째를 받겠다는 의미 //
  @Query(
    value = 
    "SELECT  " +
    "B.board_number AS boardNumber, " +
    "B.title AS title, " +
    "B.content AS content, " +
    "B.write_datetime AS writeDatetime, " +
    "B.writer_email AS writerEmail, " +
    "U.nickname AS writerNickname, " +
    "U.profile_image AS writerProfileImage " +
    "FROM board AS B " +
    "INNER JOIN user AS U " +
    "on B.writer_email = U.email " +
    "WHERE board_number = ?1 ",
    nativeQuery = true
  )
  GetBoardResultSet getBoard(Integer boardNumber);

}

File: board-back-nest/src/modules/data-access/repository/board.repository.ts
import { Injectable, Logger } from "@nestjs/common";
import { InjectRepository } from "@nestjs/typeorm";
import { BoardEntity } from "../entities";
import { DataSource, Repository } from "typeorm";
import { PostBoardRequestDto } from "modules/board/dto/request";
import { ResponseDto } from "types/classes";
import { nowDatetime } from "utils";
import { GetBoardResultSet } from "../entities/result-set";

@Injectable()
export default class BoardRepository {

  private readonly logger = new Logger('Board Repository');

  constructor(
    @InjectRepository(BoardEntity)
    private readonly repository: Repository<BoardEntity>,
    private readonly dataSource: DataSource
  ) {}

  create({ title, content } :PostBoardRequestDto, writerEmail:string) {
    try {
      const boardEntity = this.repository.create({
        title,
        content,
        writeDatetime: nowDatetime,
        favoriteCount: 0,
        commentCount: 0,
        viewCount: 0,
        writerEmail
      });
      return boardEntity;
    } catch (exception) {
      this.logger.error(exception.message);
      ResponseDto.databaseError();
    }
  }

  async save(boardEntity: BoardEntity) {
    try {
      return await this.repository.save(boardEntity);
    } catch (exception) {
      this.logger.error(exception.message);
      ResponseDto.databaseError();
    }
  }

  async getBoard(boardNumber: number) {
    try {
      const resultSet = await this.dataSource
                .createQueryBuilder()
                .select("B.board_number", "boardNumber")
                .addSelect("B.title", "title")
                .addSelect("B.content", "content")
                .addSelect("B.write_datetime", "writeDatetime")
                .addSelect("B.writer_email", "writerEmail")
                .addSelect("U.nickname", "writerNickname")
                .addSelect("U.profile_image", "writerProfileImage")
                .from("board", "B")
                .innerJoin("user", "U", "B.writer_email = U.email")
                .where("B.board_number = :boardNumber", { boardNumber })
                .getRawOne();
            return resultSet as GetBoardResultSet;
    } catch (exception) {
      this.logger.error(exception.message);
      ResponseDto.databaseError();
    }
  }
  
}
